buildtools/gradle/gradle.go

```diff
@@ -160,7 +160,7 @@ func (s ShellCommand) DependencyTasks() ([]string, error) {
 //go:generate bash -c "genny -in=$GOPATH/src/github.com/fossas/fossa-cli/graph/readtree.go gen 'Generic=Dependency' | sed -e 's/package graph/package gradle/' > readtree_generated.go"
 
 func ParseDependencies(stdout string) ([]Dependency, map[Dependency][]Dependency, error) {
-	r := regexp.MustCompile("^([ `+\\\\|-]+)([^ `+\\\\|-].+)$")
+	r := regexp.MustCompile(`^((?:[ |\|]{1,1}    )*(?:[\\+|\\]{1,1}--- ))(.*)$`)
```

Comment:
```suggestion
	r := regexp.MustCompile(`^([ +\\|\x60-]{2,})([^ +\\|\x60-].+)$`)
```
We can just make sure that there are at least two characters at the beginning of the line that can't be a dependency name. Haven't tried this with the complete integration tests, but the the Gradle tests in this PR are currently failing.

Reply:
This will only work until someone accidentally uses a configuration name that matches the new regex.

`go test -count=1 -run TestGradleIntegration  ./...` succeeds on my local machine.  I assume there is something wrong with the CI system. See also the lower coverage of mercurial/svn, php, python/pip, etc.

Reply:
It's sort of unlikely someone will name something with more than one "-" or "/" at the beginning, but lest we go down a rabbit hole about this needlessly, I'm okay with the idea behind your regex, with a couple of cleanups as shown in my suggestion above. In particular, the quantifier `{1,1}` is essentially meaningless and redundant, and in the two character sets the pipe character appears twice (as `|` and `\|`). You can just have it there once as `|`. And to keep things easier to read, we'll express the four consecutive spaces using ` {4}` please. Thank you!